resources/schema.json

```diff
@@ -334,20 +352,89 @@
-                      ]
+                      ],
+                      "not":
+                      {
+                        "oneOf": [
+                          {
+                            "required": [
+                              "source"
+                            ]
+                          },
+                          {
+                            "required": [
+                              "sourceExpression"
+                            ]
+                          }
+                        ]
+                      }
                     },
                     {
                       "required": [
                         "source"
-                      ]
+                      ],
+                      "not":
+                      {
+                        "oneOf": [
+                          {
+                            "required": [
+                              "variables"
+                            ]
+                          },
+                          {
+                            "required": [
+                              "sourceExpression"
+                            ]
+                          }
+                        ]
+                      }
                     },
                     {
                       "required": [
                         "sourceExpression"
-                      ]
+                      ],
+                      "not":
+                      {
+                        "oneOf": [
+                          {
+                            "required": [
+                              "variables"
+                            ]
+                          },
+                          {
+                            "required": [
+                              "source"
+                            ]
+                          }
+                        ]
+                      }
+                    },
+                    {
+                      "required": [
+                        "variables",
+                        "sourceExpression"
+                      ],
+                      "not":
+                      {
+                        "required": [
+                          "source"
+                        ]
+                      }
+                    },
+                    {
+                      "required": [
+                        "variables",
+                        "source"
+                      ],
+                      "not":
+                      {
+                        "required": [
+                          "sourceExpression"
+                        ]
+                      }
                     }
                   ],
-                  "errorMessage": "property.binding ${0/type} requires variables, sourceExpression or source"
+                  "errorMessage": "property.binding ${0/type} exclusively requires variables, sourceExpression, source, (sourceExpression and variables), or (source and variables)"
```

Comment:
Reads really difficult in one sentence... The error is thrown once _all_ rules don't match, so when nothing is set. So maybe the simple `requires variables, sourceExpression or source` is still valid here. 

Maybe we can even move the other cases into another rule, making it explicit that some combinations are not allowed?

Reply:
I adjusted the wording and will invest in simplifying the rule

Reply:
I added a test case for the validation, which also shows how the `not` clauses help to catch overloaded configurations (meaning too many binding parameters given). 

The rule can be made shorter IMO by using some anyOf / oneOf combinations, but I think the rule as is is easier to understand, since it is basically a verbose list of what combinations we allow. 

So therefore with the added test case I would propose to keep it.

One alternative: we could not spot the overloaded case (and remove the test case) and then could just simply not check for that. That would allow us to delete the `not` clauses and switch to `anyOf`.

Reply:
cool!